src/routes/v2/authenticatedSites/media.ts

```diff
@@ -0,0 +1,419 @@
+    never,
+    { page: number; limit: number },
+    { userWithSiteSessionData: UserWithSiteSessionData }
+  > = async (req, res) => {
+    const { userWithSiteSessionData } = res.locals
+
+    const { directoryName } = req.params
+    const { page, limit } = req.query
+    const {
+      directories,
+      files,
+    } = await this.mediaDirectoryService.listMediaDirectoryContent(
+      userWithSiteSessionData,
+      {
+        directoryName,
+        page,
+        limit,
+        search: "",
+      }
+    )
+
+    return res.status(200).json([...directories, ...files])
+  }
+
+  // List files within a media directory
+  listMediaDirectoryFiles: RequestHandler<
+    { directoryName: string },
+    { files: MediaFileOutput[]; total: number },
+    never,
+    { page: number; limit: number; search: string },
+    { userWithSiteSessionData: UserWithSiteSessionData }
+  > = async (req, res) => {
+    const { userWithSiteSessionData } = res.locals
+
+    const { directoryName } = req.params
+    const { page, limit, search } = req.query
+
+    const {
+      files,
+      total,
+    } = await this.mediaDirectoryService.listMediaDirectoryContent(
+      userWithSiteSessionData,
+      {
+        directoryName,
+        page,
+        limit,
+        search,
+      }
+    )
+
+    return res.status(200).json({ files, total })
+  }
+
+  // List subdirectories within a media directory
+  listMediaDirectorySubdirectories: RequestHandler<
+    { directoryName: string },
+    { directories: MediaDirOutput[] },
+    never,
+    { page: number },
+    { userWithSiteSessionData: UserWithSiteSessionData }
+  > = async (req, res) => {
+    const { userWithSiteSessionData } = res.locals
+
+    const { directoryName } = req.params
+    const { page } = req.query
+
+    const {
+      directories,
+    } = await this.mediaDirectoryService.listMediaDirectoryContent(
+      userWithSiteSessionData,
+      {
+        directoryName,
+        page,
+        limit: 0,
+        search: "",
+      }
+    )
+
+    return res.status(200).json({ directories })
+  }
+
+  // Create a new media directory
+  createMediaDirectory: RequestHandler<
+    never,
+    { newDirectoryName: string },
+    { newDirectoryName: string; items: string[] },
+    never,
+    {
+      userWithSiteSessionData: UserWithSiteSessionData
+      githubSessionData: GithubSessionData
+    }
+  > = async (req, res) => {
+    const { userWithSiteSessionData, githubSessionData } = res.locals
+
+    const { error } = CreateMediaDirectoryRequestSchema.validate(req.body)
+    if (error) throw new BadRequestError(error.message)
+
+    const { newDirectoryName, items } = req.body
+    const createResp = await this.mediaDirectoryService.createMediaDirectory(
+      userWithSiteSessionData,
+      githubSessionData,
+      {
+        directoryName: newDirectoryName,
+        objArray: items,
+      }
+    )
+
+    return res.status(200).json(createResp)
```

Comment:
why did we omit the `next()` call here?

Reply:
The `next()` call as I understand it is to let express know to continue the execution by matching other routes, which is mainly used for middleware. In this case, we don't actually need this `next()` call since this is the final route and we want the execution to end here.

Not too sure why it was added in the past, probably a copy-paste from somewhere, but verified that this is not necessary (especially since we already send a status 200).